front/components/data_source/RequestOrAddDataFromDataSourceModal.tsx

```diff
@@ -0,0 +1,52 @@
+import { Button, PlusIcon } from "@dust-tt/sparkle";
+import type { DataSourceType, WorkspaceType } from "@dust-tt/types";
+import { useState } from "react";
+
+import { ConnectorPermissionsModal } from "@app/components/ConnectorPermissionsModal";
+import { RequestDataSourceModal } from "@app/components/data_source/RequestDataSourceModal";
+import { useConnector } from "@app/lib/swr/connectors";
+
+interface RequestOrAddDataFromDataSourceModalProps {
+  owner: WorkspaceType;
+  dataSource: DataSourceType;
+}
+
+export function RequestOrAddDataFromDataSourceModal({
+  owner,
+  dataSource,
+}: RequestOrAddDataFromDataSourceModalProps) {
+  const [showConnectorPermissionsModal, setShowConnectorPermissionsModal] =
+    useState(false);
+  const { connector } = useConnector({
```

Comment:
Looking at the hook and the endpoint, I think @flvndvd wanted to say that if the datasource has no connector, it will return an error.

I see that you are only rendering this modal if the datasource is managed but it's a (little) bit brittle.

(nit) Suggestions for improvements :
- refactor the hook to:
  - add `disabled` support (see other hooks)
  - pass `DataSourceType` directly instead of just the id.
  - disable (return error ?) if the `dataSource.connectorId` is null.

WDYT ?

Reply:
Should be good to go now 😇